tools/serve/serve.py

```diff
@@ -947,8 +951,9 @@ def build_config(logger, override_path=None, config_cls=ConfigBuilder, **kwargs)
     if enable_http2:
         rv._default["ports"]["h2"] = [9000]
 
-    if kwargs.get("webtransport_h3"):
-        rv._default["ports"]["webtransport-h3"] = [11000]
+    # Provide a default port number for WebTransport over HTTP/3 even if it is not
+    # enabled so that port substitutions work.
+    rv._default["ports"]["webtransport-h3"] = [11000]
```

Comment:
Look at `ConfigBuilder` around like 861, I think we can just put the default as "auto" in that.

Reply:
Thank you! I wasn't aware "auto" selects an available` port automatically by opening a socket. That's much better. Done.